db/pushdb.go

```diff
@@ -145,6 +146,37 @@ func (f *pushDatabaseOpts) AddPushServiceProviderToService(service string,
 	}
 	f.dblock.Lock()
 	defer f.dblock.Unlock()
+
+	/*
+	 * #198:
+	 * 2017 Victor Lang
+	 * Before add a new psp to service, try to verify there is no redundant psp in same type exists under the same service.
+	 * Currently, redundant psp to service will result in problem in subscribe and push later.
+	 */
+	expsps, err := f.db.GetPushServiceProvidersByService(service)
+	if err != nil {
+		return fmt.Errorf("Error querying psp with name: %v", err)
+	}
+
+	for _, pspitem := range expsps {
+		pushpsp, perr := f.db.GetPushServiceProvider(pspitem)
+		if perr != nil {
+			return fmt.Errorf("Error to retrieve psp with name: %v", perr)
```

Comment:
Still not enough information to fix this: `"Error retrieving PSP (of service %s) with id %s: %v", service, pushpsp, perr`